aip/0194.md

```diff
@@ -45,18 +46,21 @@ The following codes **should not** be retried for any request:
 
 - `OK`: The request succeeded.
 - `CANCELLED`: An application can cancel a request, which **must** be honored.
-- `DEADLINE_EXCEEDED`: An application can set a deadline, which **must** be honored.
-- `INVALID_ARGUMENT`: Retrying a request with an invalid argument will never succeed.
-- `DATA_LOSS`: This is an unrecoverable error and **must** immediately be surfaced to the application.
+- `DEADLINE_EXCEEDED`: An application can set a deadline, which **must** be
```

Comment:
Isn't part of the point of setting a deadline to allow the client to retry if they get caught in the latency tail? See go/tott-531 for a practical example of this.

Reply:
Our assessment is, "We can not really know why the client set a deadline." That is one possible reason, but by no means the only one. At one point, our first-party client libraries hard-coded deadlines with this expectation and it caused us numerous issues.

(Another reason is, "This is not valuable to me if it does not show up in N seconds.")

Reply:
Ah, this makes more sense if we're referring to the client library rather than the application itself. Clearly, the request should not be *automatically* retried by the client library, but the application should be free to retry if it would be useful.

Reply:
Right, this is about setting up _automatic_, customer-invisible retries.